### PR TITLE
Key gear item detail-screen: Photo Carousel and more

### PR DESCRIPTION
**Pull request by hedvigoscar**
This PR got a bit out of hand due to the different blockers and branches that had to be created around them.

Once it's reviewed and merged however I think we're back on track

### CODE REVIEW

File: app/src/debug/java/com/hedvig/app/MockCreateKeyGearItemViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                        CreateKeyGearItemMutation.CreateKeyGearItem(
+                            "KeyGearItem",
+                            CreateKeyGearItemMutation.CreateKeyGearItem.Fragments(
+                                MockKeyGearItemDetailViewModel.items["234"]!!
+                            )
+                        )
+                    )
+                )
+            }
+            else -> {
```
Comment (FredrikAreschoug) [inline app/src/debug/java/com/hedvig/app/MockCreateKeyGearItemViewModel.kt]: I don't like `else` in when statements. I think it removes the awesome functionality of not compiling when you add a new one

File: app/src/debug/res/layout/activity_development.xml [hunk 2]
```diff
@@ -82,14 +82,23 @@
...
             android:text="Open an logged in with Welcome Screen" />
 
+        <Button
+            android:id="@+id/openLoggedInWithoutWelcome"
+            style="@style/HedvigButton.Purple"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center_horizontal"
+            android:layout_marginTop="@dimen/base_margin_double"
+            android:text="Open an logged in without Welcome Screen" />
```
Comment (FredrikAreschoug) [inline app/src/debug/res/layout/activity_development.xml]: Add a todo if the text key doesn't exists

File: app/src/main/java/com/hedvig/app/feature/keygear/ui/createitem/Category.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+val KeyGearItemCategory.label: String
+    get() =
+        when (this) {
+            KeyGearItemCategory.COMPUTER -> "Dator"
+            KeyGearItemCategory.PHONE -> "Mobiltelefon"
+            KeyGearItemCategory.TV -> "TV"
+            KeyGearItemCategory.JEWELRY -> "Smycke"
+            KeyGearItemCategory.SOUND_SYSTEM -> "Ljudanläggning"
+            else -> ""
```
Comment (FredrikAreschoug) [inline app/src/main/java/com/hedvig/app/feature/keygear/ui/createitem/Category.kt]: Same here

File: app/src/main/java/com/hedvig/app/feature/keygear/ui/itemdetail/KeyGearItemDetailActivity.kt [hunk 4]
```diff
@@ -2,13 +2,107 @@ package com.hedvig.app.feature.keygear.ui.itemdetail
...
+
+        root.useEdgeToEdge()
+        initializeToolbar()
+        initializePhotos(
+            intent.getStringExtra(FIRST_PHOTO_URL),
+            intent.getSerializableExtra(CATEGORY) as? KeyGearItemCategory
+        )
+
+        model.data.observe(this) { data ->
+            data?.let { bind(it) }
```
Comment (FredrikAreschoug) [inline app/src/main/java/com/hedvig/app/feature/keygear/ui/itemdetail/KeyGearItemDetailActivity.kt]: shouldn't we log if it is null?

File: app/src/main/res/layout/create_key_gear_item_new_photo.xml [hunk 5]
```diff
@@ -7,6 +7,7 @@
     android:background="@drawable/background_rounded_corners_ripple"
     android:backgroundTint="@color/violet_100"
     android:clickable="true"
+    android:elevation="2dp"
```
Comment (FredrikAreschoug) [inline app/src/main/res/layout/create_key_gear_item_new_photo.xml]: :/

File: app/src/main/res/layout/create_key_gear_item_category.xml [hunk 6]
```diff
@@ -7,6 +7,7 @@
     android:layout_height="wrap_content"
     android:background="@drawable/background_create_key_gear_item_category"
     android:clickable="true"
+    android:elevation="2dp"
```
Comment (FredrikAreschoug) [inline app/src/main/res/layout/create_key_gear_item_category.xml]: :/

File: app/src/main/res/layout/create_key_gear_item_photo.xml [hunk 7]
```diff
@@ -5,4 +5,5 @@
     android:layout_width="@dimen/create_key_gear_add_photo_height"
     android:layout_height="@dimen/create_key_gear_add_photo_height"
     android:contentDescription="@null"
+    android:elevation="2dp"
```
Comment (FredrikAreschoug) [inline app/src/main/res/layout/create_key_gear_item_photo.xml]: :/

File: app/src/main/java/com/hedvig/app/ui/view/PagerIndicatorView.kt [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+                val nItems = v.adapter?.itemCount ?: return
+                for (_i in 1..nItems) {
+                    val indicator = inflater.inflate(R.layout.pager_indicator_view, this, true)
+                    (indicator as? ImageView)?.let { image ->
+                        image.drawable.mutate().setTint(context.compatColor(R.color.blur_white))
+                    }
+                }
+            }
+        }
+
```
Comment (FredrikAreschoug) [inline app/src/main/java/com/hedvig/app/ui/view/PagerIndicatorView.kt]: mm UI programming

File: app/src/debug/java/com/hedvig/app/MockCreateKeyGearItemViewModel.kt (hunk 1 above)
Comment (hedvigoscar) [reply app/src/debug/java/com/hedvig/app/MockCreateKeyGearItemViewModel.kt]: I agree!

I'm intentionally being a little bit lax with the code in the Mock-files, though 😉

File: app/src/debug/res/layout/activity_development.xml (hunk 2 above)
Comment (hedvigoscar) [reply app/src/debug/res/layout/activity_development.xml]: Files in the `debug`-folder are allowed to have lower standards intentionally 😉

File: app/src/main/java/com/hedvig/app/feature/keygear/ui/itemdetail/KeyGearItemDetailActivity.kt (hunk 4 above)
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/keygear/ui/itemdetail/KeyGearItemDetailActivity.kt]: There will always be one `null` emitted when the ViewModel is created, so I think we should not 🙂

File: app/src/main/java/com/hedvig/app/ui/view/PagerIndicatorView.kt (hunk 8 above)
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/ui/view/PagerIndicatorView.kt]: Good times!

**Comment by hedvigoscar** [comment]
@FredrikAreschoug Fixed!

**Pull request merged**